test/users.js

```diff
@@ -0,0 +1,142 @@
+const chai = require('chai'),
+  server = require('./../app'),
+  dictum = require('dictum.js'),
+  sessionManager = require('./../app/services/sessionManager'),
+  should = chai.should();
+
+describe('users', () => {
+  describe('users sessions', () => {
+    it('should fail because of invalid email', done => {
+      chai
+        .request(server)
+        .post('/users/sessions')
+        .send({ email: 'invalid', password: '12345678' })
+        .catch(err => {
+          err.should.have.status(400);
+          err.response.should.be.json;
+          err.response.body.should.have.property('error');
+        })
+        .then(() => done());
+    });
+  });
+
+  describe('users create', () => {
+    it('creation OK', done => {
+      chai
+        .request(server)
+        .post('/users')
+        .send({
+          firstname: 'Alan',
+          lastname: 'Rinaldi',
+          email: 'alan.rinaldi@wolox.com.ar',
+          password: '12345678'
+        })
+        .then(res => {
+          res.should.have.status(201);
+          dictum.chai(res);
+        })
+        .then(() => done());
+    });
+    it('should fail because name is missing', done => {
+      chai
+        .request(server)
+        .post('/users')
+        .send({
+          lastname: 'Rinaldi',
+          email: 'alan.rinaldi@wolox.com.ar',
+          password: '1234'
+        })
+        .catch(err => {
+          err.should.have.status(400);
```

Comment:
How do you know if the error is caused by the name being missing if you dont validate anything other than the status?